### PR TITLE
Feat: New homepage design

### PR DESCRIPTION
**Pull request by yangkb09**
## description
in this PR, i implement the new homepage design based off @catherineymgui's [figma design](https://www.figma.com/design/eL7u2QrvLykWRLeUELiNNy/%5BGA%5D-Explore-Metrics---Master-File?node-id=1208-15616&node-type=instance&m=dev)

### old homepage (first time user)
Screenshot 2024-10-25 at 4 12 21 PM

### new homepage (first time user)
Screenshot 2024-10-25 at 4 10 47 PM

### old homepage (existing user)
Screenshot 2024-10-25 at 4 33 25 PM (2)

### new homepage (existing user)
Screenshot 2024-10-25 at 4 35 07 PM (2)

### new homepage (first time user)
- first time user will land directly on this page, only seeing the "billboard"
- format learn more and let’s start buttons
- hide recent metrics and bookmarks headers (and horizontal line)

### new homepage (recent explorations)
- grid layout
- new card UI
  - truncate singular label with long name
  - truncate multiple long labels after 3 lines
  - selecting a card takes you to the metric
- display only 3 recent metrics, can click "show more" button to show additional recent metrics

### bookmarks
- same as recent explorations:
  - grid layout
  - new card UI
- position delete button in line with data source line, but off to the right
- don't show section if no bookmarks
- collapse section by default
- expand section by clicking on arrow icon
- when expanded, arrow icon toggles up

### functionality
- select metric: recent metric card with "select metric" as the metric name is no longer created when users who only go to the select metric page (page showing lists of metrics) but do not click into a metric

### refactor
- refactored bookmarks and recent metrics into functional components

## future improvements:
- add unit tests
- all the discussion comments below
- new card UI: 
  - sometimes when multiple long labels get truncated, the labels end with a `|` => expected behavior is it should not end in a `|` but rather `-end-of-label...`
  - responsive card height to account for multiple labels spanning more than 3 lines OR implement hovering on labels to view the truncated labels 
- bookmarks: toggle icon to upward arrow when expanded

ref [#91815](https://github.com/grafana/grafana/issues/91815)

### CODE REVIEW

**Comment by itsmylife** [comment]
BugBash: Button hover state needs some spacing 
image

**Comment by itsmylife** [comment]
BugBash: Really long metrics names are looking crazy on the card. I think they need some truncating too. 
image

**Comment by itsmylife** [comment]
BugBash: (some more findings)
- Pipe color is different than the design
- In the design it says "Last metric" but the implementation uses "Metric"
- Metric and labels should not have space in between

**Comment by zhehao-grafana** [comment]
I assume the 3 additional recent metrics will get fixed by _(1) Show more button OR stop at 3 cards_
Screenshot by Dropbox Capture

**Comment by NWRichmond** [comment]
I just found this Saga design system resource: https://grafana.com/developers/saga/templates/lists-of-objects/#tiled-cards. With that info in mind, I believe there's an opportunity to replace some of the CSS grid work with the `<Grid />` component (see [Saga](https://grafana.com/developers/saga/templates/grid-page) and [Storybook](https://developers.grafana.com/ui/latest/index.html?path=/docs/general-layout-grid--docs)). With `columns`, `minColumnWidth`, `gap`, and `alignItems` props, it sounds like it could do everything that's needed to achieve the design.

This might not be crucial to the initial implementation, but when the window is resized and becomes narrow, the right & left columns become cut off:

exploremetrics homepage columns with narrow window

Maybe the `<Grid />` component would help with this 🤔

File: public/app/features/trails/DataTrailCard.tsx
Comment (NWRichmond) [inline public/app/features/trails/DataTrailCard.tsx:77]: Not sure if this matters, but the designs have "Data source:" instead of "Datasource:"
```suggestion
          <div className={styles.secondaryFont}>Data source: </div>
```

**Comment by yangkb09** [comment]
i'm not sure why i have changes in `go.work.sum` and `.betterer.results`, this happened after i merged main into my branch to fix a few merge conflicts

File: public/app/features/trails/DataTrailCard.tsx
Comment (yangkb09) [reply public/app/features/trails/DataTrailCard.tsx:77]: ah great catch!

**Comment by yangkb09** [comment]
> BugBash: (some more findings)
> 
> * Pipe color is different than the design
> * In the design it says "Last metric" but the implementation uses "Metric"
> * Metric and labels should not have space in between

re: the last 2 bullets, @catherineymgui and i agreed on some changes during a meeting - what we agreed on is reflected in the current homepage so it's okay to disregard those 2 discrepancies between homepage and figma design :)

**Comment by yangkb09** [comment]
i did notice a bug while testing - this behavior was happening prior to my changes too. basically:

click onto Metric A => use trail crumbs to go back to metric select view => select Metric B => click into homepage (via top nav bar) => only Metric B shows on recent metrics

do we want both Metric A and Metric B to show in the recent metrics section?

File: public/app/features/trails/DataTrail.tsx
Comment (itsmylife) [inline public/app/features/trails/DataTrail.tsx:141]: It's quite a nice solution to prevent saving here in one place. But without a metric we also prevent saving a search with bunch of filters and time range applied. I don't think we want that. Please verify this behavior is the desired one. If not you can check the history step count and if it is just one that means it is a fresh selection and just don't save.

File: public/app/features/trails/DataTrailBookmarks.tsx
Comment (itsmylife) [inline public/app/features/trails/DataTrailBookmarks.tsx:19]:
```suggestion
  const styles = useStyles2(getStyles);

  return (
```

Comment (itsmylife) [reply public/app/features/trails/DataTrailBookmarks.tsx:19]: Just a new line between return statement.

File: public/app/features/trails/DataTrailCard.tsx
Comment (itsmylife) [inline public/app/features/trails/DataTrailCard.tsx:146]: Maybe there is a mistake in the UI design. @catherineymgui Could you please check the design. 
Having an eslint rule for borderRadius is weird enough and having it disabled is even weirder :D
But let's note this down to somewhere to check after merging the PR. Not a blocker. But it would be nice to understand why we have that rule in the first place and how we can play with it nicely.

File: public/app/features/trails/DataTrailsRecentMetrics.tsx
Comment (itsmylife) [inline public/app/features/trails/DataTrailsRecentMetrics.tsx:22]: If recentMetrics length is 0 then let's return null immediately.

File: public/app/features/trails/DataTrailsRecentMetrics.tsx
Comment (itsmylife) [inline public/app/features/trails/DataTrailsRecentMetrics.tsx:21]:
```suggestion
  };

  return (
```

File: public/app/features/trails/DataTrailBookmarks.tsx
Comment (itsmylife) [inline public/app/features/trails/DataTrailBookmarks.tsx:16]:
```suggestion
}

export function DataTrailsBookmarks({ model, onDelete }: Props) {
```

**Review by itsmylife — COMMENTED** [review]
In general it looks good and works nice. There are just some nitpicks. 
Great job 🚀

**Comment by yangkb09** [comment]
> I just found this Saga design system resource: https://grafana.com/developers/saga/templates/lists-of-objects/#tiled-cards. With that info in mind, I believe there's an opportunity to replace some of the CSS grid work with the `<Grid />` component (see [Saga](https://grafana.com/developers/saga/templates/grid-page) and [Storybook](https://developers.grafana.com/ui/latest/index.html?path=/docs/general-layout-grid--docs)). With `columns`, `minColumnWidth`, `gap`, and `alignItems` props, it sounds like it could do everything that's needed to achieve the design.
> 
> This might not be crucial to the initial implementation, but when the window is resized and becomes narrow, the right & left columns become cut off:
> 
> exploremetrics homepage columns with narrow window
> 
> Maybe the `<Grid />` component would help with this 🤔

@NWRichmond this is very cool! noted - will look into if we decide to dedicate a bit more time to polishing things up :)

**Comment by catherineymgui** [comment]
This is looking really good!

Feedback:

1. Currently light mode looks like this. Would check the tokens! I added a figma design for light mode too: https://www.figma.com/design/eL7u2QrvLykWRLeUELiNNy/%5BGA%5D-Explore-Metrics---Master-File?node-id=1522-8195&t=XFvfzGG6x4adz7as-1 

Screenshot 2024-10-25 at 13 30 52

2. The hover state on the arrow is overlapping with the text
Screenshot 2024-10-25 at 13 32 24

3. I know you implemented the padding to spec, but I have made some mistakes in them. They should all be multiples of 4/8px.
So the gap should be 32 instead of 31
Screenshot 2024-10-25 at 13 46 08

A the bottom margin should be 24 instead of 20:
Screenshot 2024-10-25 at 13 45 18

4. I'm not sure if we can solve for this, but noticed that not only does the page jump left (I assume due to the scroll bar), but also a bit to the top when the bookmark section is expanded

https://github.com/user-attachments/assets/f70731fc-51f6-465d-943c-9490e7407192

5. I noticed explorations without a metric selected aren't saved in recents anymore - but they are also not showing if labels are added. See [this PR](https://github.com/grafana/grafana/issues/93073) for more info 

As a fyi I feel like the sections are a bit tight ever since we added the bookmarks (I know this is by design) but I will think about this a bit more

File: public/app/features/trails/DataTrailCard.tsx
Comment (catherineymgui) [reply public/app/features/trails/DataTrailCard.tsx:146]: Which part of the design is this? Not sure where there should be 4px border radius

Comment (yangkb09) [reply public/app/features/trails/DataTrailCard.tsx:150]: in the new card UI, this is to make the top left and top right corners of the cards rounded, while keeping the bottom left and bottom right of the cards a right angle. i'll demonstrate what i mean in the weekly today!

Comment (NWRichmond) [reply public/app/features/trails/DataTrailCard.tsx:150]: Can this be accomplished with `theme.shape.radius[something]`?

File: public/app/features/trails/DataTrail.tsx
Comment (yangkb09) [reply public/app/features/trails/DataTrail.tsx:141]: ah great catch, thank you! will address when pairing with brendan tomorrow

**Comment by yangkb09** [comment]
> This is looking really good!
> 
> Feedback:
> 
> 1. Currently light mode looks like this. Would check the tokens! I added a figma design for light mode too: https://www.figma.com/design/eL7u2QrvLykWRLeUELiNNy/%5BGA%5D-Explore-Metrics---Master-File?node-id=1522-8195&t=XFvfzGG6x4adz7as-1
> 
> Screenshot 2024-10-25 at 13 30 52
> 2. The hover state on the arrow is overlapping with the text
> 
> Screenshot 2024-10-25 at 13 32 24
> 3. I know you implemented the padding to spec, but I have made some mistakes in them. They should all be multiples of 4/8px.
>    So the gap should be 32 instead of 31
> 
> Screenshot 2024-10-25 at 13 46 08
> A the bottom margin should be 24 instead of 20: Screenshot 2024-10-25 at 13 45 18
> 
> 4. I'm not sure if we can solve for this, but noticed that not only does the page jump left (I assume due to the scroll bar), but also a bit to the top when the bookmark section is expanded
> 
>  Screen.Recording.2024-10-25.at.13.51.01.mov 
> 5. I noticed explorations without a metric selected aren't saved in recents anymore - but they are also not showing if labels are added. See [this PR](https://github.com/grafana/grafana/issues/93073) for more info
> 
> As a fyi I feel like the sections are a bit tight ever since we added the bookmarks (I know this is by design) but I will think about this a bit more

@catherineymgui so far, i implemented:

- [ x ] #2
- [ x ] #3

File: public/app/features/trails/DataTrailCard.tsx
Comment (yangkb09) [reply public/app/features/trails/DataTrailCard.tsx:146]: basically, the way i have the card set up is that part A is one section and part B is a separate section. i did this to implement the different background colors between parts A and B. so part A needs to have it's top left and top right corners rounded to 4px, but the bottom left & right corners need to be a right angle (0px) since that's where part A meets part B

part B will have the top left and right corners as right angles, and the bottom left/right corners as rounded 4px
462A9AC1-A0D0-457D-ACD2-46164FA25EB2

if that doesn't make sense i can clarify in the daily tomorrow :)

Comment (catherineymgui) [reply public/app/features/trails/DataTrailCard.tsx:146]: Yeah this makes sense. I believe in my designs it's 2px corner radius instead of 4px tho (top left and right)?

For more context, the team updated all the corner radii across Grafana from 4px to 2px sometime ago.

File: public/app/features/trails/DataTrail.tsx
Comment (bohandley) [inline public/app/features/trails/DataTrail.tsx:141]: Check for those filters with this special Scenes variable lookup 👀 

```suggestion
    const saveRecentTrail = () => {
      const filtersVariable = sceneGraph.lookupVariable(VAR_FILTERS, this);
      const hasFilters = filtersVariable instanceof AdHocFiltersVariable && filtersVariable.state.filters.length > 0;

      return ((this.state.metric || hasFilters) 
      && getTrailStore().setRecentTrail(this))};```

Comment (bohandley) [reply public/app/features/trails/DataTrail.tsx:141]: The variable lookup is interesting 🤔 You might think the filters are stored in the state but this `VAR_FILTERS` variable is one way that Scenes can access stored data across all the Scenes components.

File: public/app/features/trails/DataTrailCard.tsx
Comment (bohandley) [inline public/app/features/trails/DataTrailCard.tsx:29]: Woohoo! Amazing 🤘🏻 Love this truncation

**Comment by bohandley** [comment]
For the light mode issues, we can look at this file to replace any hardcoded styles and use the `theme.<thing>.<you>.<want>.<styled>`

https://github.com/grafana/grafana/blob/kat/exm-homepage/contribute/style-guides/themes.md#background-colors

**Review by bohandley — COMMENTED** [review]
Great work @yangkb09! We can look at the things I commented on tomorrow. I found nothing that hadn't already been bug bashed. Great work @grafana/observability-metrics on bug bashing!!!

We'll get this all nice and sorted tomorrow. Well done, Kat!!! 👏🏻 👏🏻 👏🏻

**Review by bohandley — APPROVED** [review]
Paired with @yangkb09 and fixed the light mode issue and the saving trail with filters. Happy to wait for Catherine and Nick's feedback but ✅ from me!

LGTM! 🚀

File: public/app/features/trails/DataTrailBookmarks.tsx
Comment (NWRichmond) [inline public/app/features/trails/DataTrailBookmarks.tsx:55]: We can simplify the JSX with an early return:

```suggestion
  if (getTrailStore().bookmarks.length === 0) {
    return null;
  }

  return (
    <>
      <div className={styles.horizontalLine} />
      <div className={css(styles.gap20, styles.bookmarkHeader, styles.bottomGap24)}>
        <div className={styles.header} style={{ marginRight: '8px' }}>
          <Trans i18nKey="trails.bookmarks.or-view-bookmarks">Or view bookmarks</Trans>
        </div>
        <IconButton
          name={toggleBookmark ? 'angle-up' : 'angle-down'}
          size="xxxl"
          aria-label="bookmarkCarrot"
          variant="secondary"
          onClick={() => setToggleBookmark(!toggleBookmark)}
        />
      </div>
      {toggleBookmark && (
        <div className={styles.trailList}>
          {getTrailStore().bookmarks.map((bookmark, index) => {
            return (
              <DataTrailCard
                key={getBookmarkKey(bookmark)}
                bookmark={bookmark}
                onSelect={() => model.onSelectBookmark(index)}
                onDelete={() => onDelete(index)}
              />
            );
          })}
        </div>
      )}
    </>
  );
```

File: public/app/features/trails/TrailStore/TrailStore.ts
Comment (NWRichmond) [inline public/app/features/trails/TrailStore/TrailStore.ts:1]: Are we using `lastModified` anywhere? If not, perhaps this change could be held back until we're using it.

**Review by NWRichmond — COMMENTED** [review]
This is coming along nicely! I'll take a look at light mode fixes shortly. In the meantime, I have some non-blocking suggestions & nits.

In addition to some feedback below on specific files, here are some thoughts about some overarching themes:
- Consider leveraging elements like [article](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article), [section](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section), etc. in place of a surrounding fragment or div.
- There are some styles (like `fontStyle: 'normal'` and `fontFamily: 'Inter'`) that don't appear to be changing anything. Removing these will reduce the size of the JS associated with our styles.
- There are some `div`s with styles that contain comments like `/* H4 */`, and I'm curious if these could actually be `<Text variant="h4">`s instead 🤔

File: public/app/features/trails/DataTrailsHome.tsx
Comment (yangkb09) [reply public/app/features/trails/DataTrailsHome.tsx:1]: i moved them to an assets folder!

**Review by NWRichmond — APPROVED** [review]
Light mode is looking beautiful!

demo light mode explore metrics homepage redesign

Nicely done, @yangkb09. Everything appears to be working as expected. What a nice glow-up for Explore Metrics!

**Comment by zhehao-grafana** [comment]
> Light mode is looking beautiful!
> 
> demo light mode explore metrics homepage redesign
> 
> Nicely done, @yangkb09. Everything appears to be working as expected. What a nice glow-up for Explore Metrics!

It is truly beautiful and thanks for the great work from Kat and Brendan!

**Comment by catherineymgui** [comment]
This is looking good based on my feedback!

Few things:
1. In this [issue](https://github.com/grafana/grafana/issues/93073) - we also mentioned changing the text of 'Select metric' to 'All metrics', for the saved recent explorations on the 'All metrics' page. if it's a quick thing, I think we could update this. Or we can leave it to another PR
2. Page jumping left - this seems to not show up in Chrome, only Arc Browser. But given that not many people use it, it's okay to ignore this
3. Based on our slack convo, we talked about a few things to improve the design of the page (since the recent metric cards are quite big, I'm finding that it's taking away from the 'Let's start') button. The proposal is to increase the size of the 'Let's start' section / card, and decrease the number of lines in the recent metric and bookmark cards. We can discuss if this should be in this PR, or another one. I will put this in slack too
Screenshot 2024-10-29 at 14 23 34

File: public/app/features/trails/TrailStore/TrailStore.ts
Comment (yangkb09) [reply public/app/features/trails/TrailStore/TrailStore.ts:1]: i tried undoing the `lastModified` changes but then the recent metrics show more functionality stopped working, so for now we need it!

**Pull request merged**
